[EXPERIMENT] message_controls: Experiment with over-message hover.

karlstolley opened the pull request:
This PR introduces experimental hover controls that do not sit within the message grid. Instead, they appear below the timestamp.

In this iteration, stars have been moved out of the hover-control group and placed in their own small column to the left of the timestamp. They are hoverable and interactive.

An issue yet to be addressed here is the likely unwanted display of hover controls appearing when the top of the message box has scrolled off screen.

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

_2025-04-16 afternoon update (note, as always, that the arrow cursor is an artifact of macOS screen recording):_


![hover-controls-2025-04-16](https://github.com/user-attachments/assets/0e2dee90-30c0-47e7-8d0c-b975ad97c211)

karlstolley committed:
message_controls: Experiment with over-message hover.

karlstolley committed:
message_controls: Show hover controls on hotkey popovers.

karlstolley committed:
message_controls: Experiment with softer shadow, borders.

alya commented:
The current version incorrectly shifts the controls up on lines that have a sender:
![Screenshot 2025-04-10 at 15 06 12](https://github.com/user-attachments/assets/8a309140-7498-4afc-b68b-241d2d880527)

alya commented:
I think it was mentioned that opening menus with the keyboard places them incorrectly, but adding a note here.


![Screenshot 2025-04-10 at 15 12 13](https://github.com/user-attachments/assets/e11b9db9-0ba4-401a-b17c-e5cda0ffa776)

alya commented:
There's an inappropriate blank space on controls shown on your own messages when you aren't allowed to edit them.

![Screenshot 2025-04-10 at 15 14 48](https://github.com/user-attachments/assets/1299bd7a-5c2a-4182-a198-c62a50862257)

karlstolley committed:
message_controls: No longer placehold icons.

Because the hover controls no longer participate in the message
grid, these reflow-prevention measures are no longer required.
In a hover context, they also placehold for icons that might not
be available to a given user (e.g., editing one's own message
when that's not allowed).

karlstolley commented:
Thanks for all of that useful feedback, @alya! I've just pushed a version with those fixes plus additional matters we've discussed at [#design > experiment: over-message hover controls](https://chat.zulip.org/#narrow/channel/101-design/topic/experiment.3A.20over-message.20hover.20controls)

alya commented:
Testing the PR, it's not handling figuring out the location of where to put these on a 1-line author-less message I just sent. When I hovered right away, the controls first appeared *blank* to the left of the timestamp, then jumped below (with buttons present), and then jumped to the left again.

We can wait to fix this until we decide whether we actually want to go with this placement.

karlstolley committed:
message_controls: Move starred controls into message body.

karlstolley committed:
message_controls: Experiment with on-message stars.

karlstolley committed:
message_box: Hold top row to sender line-height.

Because of how the template areas are defined, it's not
necessary (and actually kind of a pain) to let the top
row stretch beyond sender line-height. This allows us
to better center stars and other top-line elements.
